web/auth/auth_test.go

```diff
@@ -71,6 +71,29 @@ func getSessionID(cookies []*http.Cookie) string {
 	return ""
 }
 
+func TestInstanceBlocked(t *testing.T) {
+	// Block the instance
+	testInstance.Blocked = true
+	couchdb.UpdateDoc(couchdb.GlobalDB, testInstance)
+
+	req, _ := http.NewRequest("GET", ts.URL+"/auth/login", nil)
+	req.Host = testInstance.Domain
+
+	res, err := client.Do(req)
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusPaymentRequired, res.StatusCode)
+
+	// Trying with a Accept: text/html header to simulate a browser
+	req.Header.Add("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8")
+	res2, err := client.Do(req)
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusPaymentRequired, res2.StatusCode)
```

Comment:
Can you also check that we don't have a blank page? Maybe just just look at a title in the body